File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        
        
        costs.sort()
        
        cnt = 0 
        for i,c in enumerate(costs):
            
            coins -=c
            if coins ==0:
                return i+1
            elif coins <0:
                return i
        return len(costs)

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.java
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        
        
        sort(costs.begin() , costs.end() ) ; 

        int ans =0 ; 
        for(auto c:costs){
            coins -= c ;
            ans++ ; 
            if(coins ==0) return ans ; 
            else if(coins < 0) return ans-1; 
        } return ans; 

    }
};